cmd/lakectl/cmd/root_test.go

```diff
@@ -71,90 +71,82 @@ func TestLakectlUnixPerm(t *testing.T) {
 			},
 			expectedURL: "http://flag-endpoint",
 		},
 		{
-			name: "using LAKECTL_CONFIG_FILE env var",
-			setup: func(t *testing.T) func() {
-				tmpFile := filepath.Join(t.TempDir(), "env_config.yaml")
-				content := "server:\n  endpoint_url: \"http://env-endpoint\"\n"
-				require.NoError(t, os.WriteFile(tmpFile, []byte(content), 0644))
-				t.Setenv("LAKECTL_CONFIG_FILE", tmpFile)
-				cfgFile = ""
-				return func() {
-					os.Unsetenv("LAKECTL_CONFIG_FILE")
-				}
+			name: "env",
+			setup: func(t *testing.T) {
+				t.Helper()
+				const content = "server:\n  endpoint_url: \"http://env-endpoint\"\n"
+				path := filepath.Join(t.TempDir(), "env_config.yaml")
+				require.NoError(t, os.WriteFile(path, []byte(content), 0644))
+				t.Setenv("LAKECTL_CONFIG_FILE", path)
 			},
 			expectedURL: "http://env-endpoint",
 		},
 		{
-			name: "using home directory default",
-			setup: func(t *testing.T) func() {
-				homeDir := t.TempDir()
-				tmpFile := filepath.Join(homeDir, ".lakectl.yaml")
-				content := "server:\n  endpoint_url: \"http://home-endpoint\"\n"
-				require.NoError(t, os.WriteFile(tmpFile, []byte(content), 0644))
-				t.Setenv("HOME", homeDir)
-				cfgFile = ""
-				return func() {
-					os.Unsetenv("HOME")
-				}
+			name: "home",
+			setup: func(t *testing.T) {
+				t.Helper()
+				const content = "server:\n  endpoint_url: \"http://home-endpoint\"\n"
+				home := t.TempDir()
+				require.NoError(t, os.WriteFile(filepath.Join(home, ".lakectl.yaml"), []byte(content), 0644))
+				t.Setenv("HOME", home)
 			},
 			expectedURL: "http://home-endpoint",
 		},
 		{
-			name: "both --config flag and env var are set, flag should win",
-			setup: func(t *testing.T) func() {
-				tmpFileFlag := filepath.Join(t.TempDir(), "config_flag.yaml")
-				flagContent := "server:\n  endpoint_url: \"http://from-flag\"\n"
-				require.NoError(t, os.WriteFile(tmpFileFlag, []byte(flagContent), 0644))
-
-				tmpFileEnv := filepath.Join(t.TempDir(), "config_env.yaml")
-				envContent := "server:\n  endpoint_url: \"http://from-env\"\n"
-				require.NoError(t, os.WriteFile(tmpFileEnv, []byte(envContent), 0644))
-
-				cfgFile = tmpFileFlag
-				t.Setenv("LAKECTL_CONFIG_FILE", tmpFileEnv)
-
-				return func() {
-					cfgFile = ""
-					os.Unsetenv("LAKECTL_CONFIG_FILE")
-				}
+			name: "flag_and_env",
+			setup: func(t *testing.T) {
+				t.Helper()
+				const flagContent = "server:\n  endpoint_url: \"http://from-flag\"\n"
+				const envContent = "server:\n  endpoint_url: \"http://from-env\"\n"
+
+				cfgFile = filepath.Join(t.TempDir(), "flag.yaml")
+				envPath := filepath.Join(t.TempDir(), "env.yaml")
+
+				require.NoError(t, os.WriteFile(cfgFile, []byte(flagContent), 0644))
+				require.NoError(t, os.WriteFile(envPath, []byte(envContent), 0644))
+
+				t.Setenv("LAKECTL_CONFIG_FILE", envPath)
 			},
 			expectedURL: "http://from-flag",
 		},
 	}
 
+	originalCfgFile := cfgFile
+	defer func() {
+		cfgFile = originalCfgFile
+		initConfig()
```

Comment:
its my bad, so you can drop this one - we no longer update the global cfg so it is not required.